### PR TITLE
Allow user to change the nuget.org feed name without breaking Nuget suggestions

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/27010

Still working on a better test.

### CODE REVIEW

File: src/Features/Core/Portable/AddImport/PackageSourceHelper.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            // Package source names are user configurable, but various operations and background tasks process
+            // only the nuget source, so we ignore the user defined name for nuget.org so we can identify it later.
+
+            var foundNugetOrg = false;
+            foreach (var packageSource in packageSources)
+            {
+                // If the user has multiple sources from nuget.org, we only need one of them to be special
+                if (!foundNugetOrg)
+                {
+                    var host = new Uri(packageSource.Source).Host;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddImport/PackageSourceHelper.cs:28]: i worry about this line throwing.  can you guarantee that 'new URI' succeeds on any string (as well as .Host)?

File: src/Features/Core/Portable/AddImport/PackageSourceHelper.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    if (host.Equals(NugetOrg, StringComparison.OrdinalIgnoreCase) ||
+                        host.EndsWith($".{NugetOrg}", StringComparison.OrdinalIgnoreCase))
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddImport/PackageSourceHelper.cs:30]: what's an example of this?

File: src/Features/Core/Portable/AddImport/PackageSourceHelper.cs (hunk 1 above)
Comment (davidwengier) [reply src/Features/Core/Portable/AddImport/PackageSourceHelper.cs:28]: Yep, will fix.

File: src/Features/Core/Portable/AddImport/PackageSourceHelper.cs (hunk 2 above)
Comment (davidwengier) [reply src/Features/Core/Portable/AddImport/PackageSourceHelper.cs:30]: This is for the standard nuget.org feed url, for which `api.nuget.org` is the host.

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/AddImport/PackageSourceHelper.cs:30]: Doc me Amadeus

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**